src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -1590,13 +1590,15 @@ def forward(
             hidden_states = self.adapter(hidden_states)
 
         if not return_dict:
-            return (hidden_states, extract_features) + encoder_outputs[1:]
+            outputs = (hidden_states, extract_features) + encoder_outputs[1:]
+            return outputs + (attention_mask,) if attention_mask is not None else outputs
 
         return Wav2Vec2BaseModelOutput(
             last_hidden_state=hidden_states,
             extract_features=extract_features,
             hidden_states=encoder_outputs.hidden_states,
             attentions=encoder_outputs.attentions,
+            attention_mask=attention_mask,
```

Comment:
We input a `torch.LongTensor` of 0/1's as the attention mask input. Currently, we're returning a boolean output for the downsampled attention mask -> IMO it makes sense to return the downsampled attention mask as a `torch.LongTensor` of 0/1's for consistency with the inputs. WDYT? Just requires you to convert this boolean tensor to a long tensor!

Reply:
Ok, I will convert it to `torch.long`.

Reply:
Perfect, thanks @gau-nernst!